CODE_STYLE.md

```diff
@@ -0,0 +1,943 @@
+
+    -- good
+    if test < 1 and do_complicated_function(test) == false or
+       seven == 8 and nine == 10 then
+
+      do_other_complicated_function() 
+      return false 
+    end
+    ```
+
+    **[[back to top]](#TOC)**
+
+
+### <a name='whitespace'>Whitespace</a>
+
+  - Use soft tabs set to 2 spaces. Tab characters and 4-space tabs result in public flogging.
+
+    ```lua
+    -- bad
+    function() 
+    ∙∙∙∙local name
+    end
+
+    -- bad
+    function() 
+    ∙local name
+    end
+
+    -- good
+    function() 
+    ∙∙local name
+    end
+    ```
+
+  - Place 1 space before opening and closing braces. Place no spaces around parens.
+
+    ```lua
+    -- bad
+    local test = {one=1}
+
+    -- good
+    local test = { one = 1 }
+
+    -- bad
+    dog.set("attr",{
+      age = "1 year",
+      breed = "Bernese Mountain Dog",
+    })
+
+    -- good
+    dog.set("attr", {
+      age = "1 year",
+      breed = "Bernese Mountain Dog",
+    })
+    ```
+
+  - Place an empty newline at the end of the file.
+
+    ```lua
+    -- bad
+    (function(global) 
+      -- ...stuff...
+    end)(self)
+    ```
+
+    ```lua
+    -- good
+    (function(global) 
+      -- ...stuff...
+    end)(self)
+         --> invisible newline here
+    ```
+
+  - Surround operators with spaces.
+
+    ```lua
+    -- bad
+    local thing=1
+    thing = thing-1
+    thing = thing*1
+    thing = 'string'..'s'
+
+    -- good
+    local thing = 1
+    thing = thing - 1
+    thing = thing * 1
+    thing = 'string' .. 's'
+    ```
+
+  - Use one space after commas.
+
+    ```lua
+    --bad
+    local thing = {1,2,3}
+    thing = { 1 , 2 , 3 }
+    thing = { 1 ,2 ,3 }
+
+    --good
+    local thing = { 1, 2, 3 }
+    ```
+
+  - Add double line breaks after top level functional blocks. Top level blocks
+    are 'requires', 'shadow-globals', 'constants', and 'functions'.
+
+    ```lua
+    -- bad
+    local x = require("x")
+    local y = require("y")
+    local insert = table.insert
+    local remove = table.remove
+    local NEVER_CHANGES = "different again"
+    local function does_cool_stuff()
+      -- totally cool stuff here
+    end
+    local function does_hot_stuff()
+      -- totally hot stuff here
+    end
+    return {
+      NEVER_CHANGES = NEVER_CHANGES,
+      cool = does_cool_stuff,
+      hot = does_hot_stuff,
+    }
+    
+    -- good
+    local x = require("x")
+    local y = require("y")
+
+
+    local insert = table.insert
+    local remove = table.remove
+
+
+    local NEVER_CHANGES = "different again"
+
+
+    local function does_cool_stuff()
+      -- totally cool stuff here
+    end
+
+
+    local function does_hot_stuff()
+      -- totally hot stuff here
+    end
+
+
+    return {
+      NEVER_CHANGES = NEVER_CHANGES,
+      cool = does_cool_stuff,
+      hot = does_hot_stuff,
+    }
+    ```
+
+  - Add a line break after multiline blocks and before `else` and `elseif` blocks.
```

Comment:
Good point, as per the Nginx/OpenResty codebases, but this is not enforced in the other examples. Should it?